Formula/producthunt-cli.rb

```diff
@@ -0,0 +1,41 @@
+class ProducthuntCli < Formula
+  desc "CLI application for the Product Hunt"
+  homepage "https://sibis.dev/"
+  version "0.1.0"
+  bottle :unneeded
+
+  if OS.mac?
+    url "https://github.com/sibis/producthunt-cli/releases/download/v0.1.0/producthunt-cli_0.1.0_Darwin_x86_64.tar.gz"
+    sha256 "45941528552dca7175e2c4d68812fb8648d6eddad374c02f43e09003419d3367"
+  elsif OS.linux?
+    if Hardware::CPU.intel?
+      url "https://github.com/sibis/producthunt-cli/releases/download/v0.1.0/producthunt-cli_0.1.0_Linux_x86_64.tar.gz"
+      sha256 "a03da77eb8b2b7e97243a4b7ce383fb6647b2b59a5b0df17e28204102102a199"
+    end
+  end
+
+  depends_on "git"
+  depends_on "zsh" => :optional
+
+  conflicts_with "svn"
+  conflicts_with "bash"
+
+  def install
+    bin.install "program"
+    ...
+  end
+
+  plist_options :startup => false
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+...
+
+  EOS
+  end
+
+  test do
+    system "#{bin}/program --version"
+    ...
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the detailed comment. Will test it out locally before the PR. Thanks